datalad/support/dataset.py

```diff
@@ -0,0 +1,60 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+This layer makes the difference between an arbitrary annex and a datalad-managed dataset.
+
+"""
+__author__ = 'Benjamin Poldrack'
+
+from os.path import join
+
+from annexrepo import AnnexRepo
+
+class Dataset(AnnexRepo):
+    """Representation of a dataset handled by datalad.
+
+    Implementations of datalad commands are supposed to use this rather than AnnexRepo or GitRepo directly,
+    since any restrictions on annexes required by datalad due to its cross-platform distribution approach are handled
+    within this class. Also an AnnexRepo has no idea of any datalad configuration needs, of course.
+
+    """
+
+    def __init__(self, path, url=None):
+        """Creates a dataset representation from path.
+
+        If `path` is empty, it creates an new repository.
+        If `url` is given, it is expected to point to a git repository to create a clone from.
+
+        Parameters
+        ----------
+
+        path : str
+               path to repository
+
+        url: str
+             url to the to-be-cloned repository.
+             valid git url according to http://www.kernel.org/pub/software/scm/git/docs/git-clone.html#URLS required.
+
+        """
+
+        super(Dataset, self).__init__(path, url)
+
+        # TODO: create proper .datalad-file (or -directory?) for marking as dataset and future use for config
```

Comment:
On Wed, 25 Feb 2015, bpoldrack wrote:

>    Well, an empty .datalad dir is a dummy, too. :)

no -- it establishes location for our meta-information (it was
envisioned to exist in some reincarnation anyways), and content of
which would also be under git control to contain handle descriptor etc

>    I just wanted to have SOMETHING to test for, since there was nothing else
>    to do for now.

so here you go -- have that directory

>    As discussion of handles progresses, we might see, that we don't need
>    anything like that at all.

we will! ;)

>    So, I will delete it completely and just test for "is_instance_of". Agree?

nope -- create .datalad/  ;) test for it
-- 
Yaroslav O. Halchenko, Ph.D.
http://neuro.debian.net http://www.pymvpa.org http://www.fail2ban.org
Research Scientist,            Psychological and Brain Sciences Dept.
Dartmouth College, 419 Moore Hall, Hinman Box 6207, Hanover, NH 03755
Phone: +1 (603) 646-9834                       Fax: +1 (603) 646-1419
WWW:   http://www.linkedin.com/in/yarik